daemon/emer-daemon.c

```diff
@@ -211,15 +219,85 @@ is_uuid (GVariant *variant)
 }
 
 static void
-uuid_from_variant (GVariant *variant,
-                   uuid_t    uuid)
+buffer_event (EmerDaemon *self,
+              GVariant   *event)
+{
+  EmerDaemonPrivate *priv = emer_daemon_get_instance_private (self);
+
+  g_variant_ref_sink (event);
+  gsize size_with_type = get_size_with_type (event);
+  gsize new_bytes_buffered = priv->num_bytes_buffered + size_with_type;
+  if (new_bytes_buffered > priv->max_bytes_buffered ||
+      /* Don't get wedged by a large event. */
+      new_bytes_buffered > MAX_REQUEST_PAYLOAD)
+    {
+      g_variant_unref (event);
+      return;
+    }
+
+  g_ptr_array_add (priv->variant_array, event);
+  priv->num_bytes_buffered = new_bytes_buffered;
+}
+
+static void
+remove_events (EmerDaemon *self,
+               gsize       num_events)
 {
-  g_variant_ref_sink (variant);
-  gsize variant_length;
-  const guchar *array =
-    g_variant_get_fixed_array (variant, &variant_length, sizeof (guchar));
-  uuid_copy (uuid, array);
-  g_variant_unref (variant);
+  EmerDaemonPrivate *priv = emer_daemon_get_instance_private (self);
+
+  for (gint i = 0; i < num_events; i++)
+    {
+      GVariant *curr_variant = g_ptr_array_index (priv->variant_array, i);
+      priv->num_bytes_buffered -= get_size_with_type (curr_variant);
+    }
+
+  if (num_events > 0)
```

Comment:
Oh fair enough; in that case though you could move the check at the beginning of the function :)